frontend/src/metabase/visualizations/visualizations/PieChart.jsx

```diff
@@ -195,13 +195,15 @@ export default class PieChart extends Component {
   };
 
   componentDidUpdate() {
-    const groupElement = ReactDOM.findDOMNode(this.refs.group);
-    const detailElement = ReactDOM.findDOMNode(this.refs.detail);
-    if (groupElement.getBoundingClientRect().width < 120) {
-      detailElement.classList.add("hide");
-    } else {
-      detailElement.classList.remove("hide");
-    }
+    requestAnimationFrame(() => {
```

Comment:
Two questions about this:
1. Does `requestAnimationFrame(f)` differ from `setTimeout(f, 0)` in this use? I guess we potentially synchronize a paint resulting from the classList change with a paint that was happening anyways?
2. Is this guaranteed to happen after the element is in the DOM and correctly sized? Or, are we just delaying the check to a point where it's more likely to have happened? Delaying and making it work the vast majority of the time is totally fine, I'm just curious if you spotted any times when it failed.

Reply:
Sure
1. `setTimeout` would work here well too. I used `requestAnimationFrame` just to allow browsers to synchronize paints as you have assumed.
2. On this step layout should be calculated already. I tested it many times in Chrome, Safari, and Firefox and it worked without a single failure